File: Heap (Priority Queue)/Minimum Operations to Exceed Threshold Value II/Minimum_Operations_to_Exceed_Threshold_Value_II.java
// Problem Name: Minimum Operations to Exceed Threshold Value II
// Problem Link: https://leetcode.com/problems/minimum-operations-to-exceed-threshold-value-ii/description/

class Solution {
    public int minOperations(int[] nums, int k) {
        int n = nums.length;
        PriorityQueue<Long> pq = new PriorityQueue<>();
        int ans = 0;
        for(int i = 0;i<n;i++) {
            pq.add(nums[i] * 1L);
        }
        while(pq.peek() < k) {
            Long x = pq.poll();
            Long y = pq.poll();
            pq.add(x * 2 + y);
            ans++;
        }
        return ans;
    }
}

File: Heap (Priority Queue)/Minimum Operations to Exceed Threshold Value II/Minimum_Operations_to_Exceed_Threshold_Value_II.cpp
// Problem Name: Minimum Operations to Exceed Threshold Value II
// Problem Link: https://leetcode.com/problems/minimum-operations-to-exceed-threshold-value-ii/description/

class Solution {
public:
    int minOperations(vector<int>& nums, int k) {
        priority_queue<long, vector<long>, greater<long>> pq(nums.begin(), nums.end());
        int ans = 0;
        while(pq.top() < k) {
            long x = pq.top();
            pq.pop();
            long y = pq.top();
            pq.pop();
            pq.push(x * 2 + y);
            ans++;
        }
        return ans;
    }
};